.github/workflows/ci.yml

```
@@ -0,0 +1,26 @@
+name: CI
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
```

```suggestion
    branches:
      - main
  pull_request:
  workflow_dispatch:
```

`main` がPRになるはずがないですね。
それから手動で実行できるように `workflow_dispatch:` を追加しました。